RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -118,16 +116,16 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const ChamberHitContainer& u
     }
 
     //change the tresholds if it's time to look for displaced mu segments
-    if(doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
-      doCollisions = false;
+    if(l_doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
```

Comment:
by construction, used_rh is a count of hits passing some selections
see a few lines above (slightly re-parsed):
```C++
    int used_rh = 0;
    for (ChamberHitContainerCIt i1 = rechits.begin(); i1 != rechits.end(); ++i1) {
      if(used[i1-ib])used_rh++;
    }
```
so, it can not be larger than rechits.size().